src/OptOut.php

```diff
@@ -0,0 +1,146 @@
+    disableStr = 'ga-disable-' + window.gaIdList[i];
+    document.cookie = disableStr + '=true; expires=Thu, 31 Dec 2099 23:59:59 UTC; path=/';
+    window[disableStr] = true;
+  }
+}
+</script>
+
+CODE;
+
+  /**
+   * @implements admin_init
+   */
+  public static function admin_init() {
+    // Register opt-out related settings.
+    register_setting(Plugin::PREFIX . '-settings', 'core_standards_optout_ids');
+    register_setting(Plugin::PREFIX . '-settings', 'core_standards_optout_script');
+    register_setting(Plugin::PREFIX . '-settings', 'core_standards_optout_active');
+
+    add_action('admin_enqueue_scripts', __CLASS__ . '::admin_enqueue_scripts');
+  }
+
+  /**
+   * @implements init
+   */
+  public static function init() {
+    if (is_admin()) {
+      return;
+    }
+
+    // Injects the Google Analytics IDs and the opt-out script in the page head
+    // section before any tracking script.
+    add_action('wp_head', __CLASS__ . '::wp_head', 5);
+
+    // Register the shortcode to call the opt-out cookie creation script.
+    add_shortcode('ga-optout-link', __CLASS__ . '::registerOptOutShortcode');
+
+    // Enqueue frontend scripts.
+    add_action('wp_enqueue_scripts', __CLASS__ . '::wp_enqueue_scripts');
+  }
+
+  /**
+   * @implements wp_head
+   */
+  public static function wp_head() {
+    if (!get_option('core_standards_optout_active')) {
+      return;
+    }
+    // Converts any non-word character into a comma and removes white-space characters
+    // (except '-') from the string to get a comma-separated list of Google Analytics IDs.
+    $optout_ids = preg_replace(['@\W^-|\n@', '@\s@'], [',', ''], esc_attr(get_option('core_standards_optout_ids')));
+  ?>
+
+<script>
+// List of Google Analytics IDs.
+window.gaIdList = <?= json_encode(explode(',', $optout_ids)) ?>;
+</script>
+  <?php
+    echo get_option('core_standards_optout_script') ?: static::$opt_out_script;
+  }
+
+  /**
+   * Renders opt-out link shortcode.
+   *
+   * @param array $atts Shortcode parameters as attributes.
+   * @param string $content Shortcode wrapped content.
+   * @return string
+   */
+  public static function registerOptOutShortcode ($atts, $content = NULL) {
+    if (get_option('core_standards_optout_active')) {
+      return '<a class="core-standards-optout-link" href="#">' . $content . '</a>';
+    }
+    else {
+      return $content;
+    }
+  }
+
+  /**
+   * Creates a submenu page in the WordPress admin "Settings" section.
+   *
+   * @implements admin_menu
+   */
+  public static function admin_menu() {
+    add_submenu_page('options-general.php', __('WordPress Core Standards Settings', Plugin::L10N), __('WordPress Core Standards', Plugin::L10N), 'manage_options', 'core-standards-settings', [__CLASS__, 'renderSettings']);
+  }
+
+  /**
+   * Renders the settings page template.
+   */
+  public static function renderSettings() {
+    Plugin::renderTemplate(['templates/settings.php']);
```

Comment:
Can we add these settings to e.g. the "Reading" page instead, so that we can simply register the settings and do not need a custom HTML output / template at all?

Reply:
@sun I did it like this because I though it would allow us to add further settings to core-standards plugin in the future.

Of course I can move this settings to the "Reading" page, but I am not sure if this is not mixing apples & oranges. Considering this, can you confirm me you want me to do so? 😉

Reply:
We generally try to avoid adding new option pages, as our plugins are usually very small.

But especially in this case it makes sense to add to existing pages, as the core-standards plugin only corrects/enhances the defaults of WordPress Core (and in an ideal world, Core would ship with everything we do in core-standards already) and doesn't really provide a very unique thing of its own.

Reply:
Note: Unless I'm mistaken, when adding to existing pages, the entire template for the administrative configuration page is obsolete.  The form elements are specified along with the call to register_setting() already.

Tried to find existing work but only found this (which isn't exactly what I mean here): https://github.com/netzstrategen/wordpress-dfp/blob/master/src/Admin.php#L43-L50